pkg/syncer/util.go

```diff
@@ -65,20 +65,29 @@ func getBoundPVs(ctx context.Context, metadataSyncer *metadataSyncInformer) ([]*
 	return boundPVs, nil
 }
 
-// getInlineMigratedVolumesInfo is a helper function for retrieving  inline PV information from Pods
-func getInlineMigratedVolumesInfo(ctx context.Context, metadataSyncer *metadataSyncInformer) (map[string]string, error) {
+// fullSyncGetInlineMigratedVolumesInfo is a helper function for retrieving  inline PV information from Pods
+func fullSyncGetInlineMigratedVolumesInfo(ctx context.Context, metadataSyncer *metadataSyncInformer, migrationFeatureState bool) (map[string]string, error) {
 	log := logger.GetLogger(ctx)
 	inlineVolumes := make(map[string]string)
 	// Get all Pods from kubernetes
 	allPods, err := metadataSyncer.podLister.List(labels.Everything())
 	if err != nil {
-		log.Errorf("getInlineMigratedVolumesInfo: failed to fetch the list of pods with err: %+v", err)
+		log.Errorf("FullSync: failed to fetch the list of pods with err: %+v", err)
 		return nil, err
 	}
+	if migrationFeatureState {
+		var err error
+		// In case if feature state switch is enabled after syncer is deployed, we need to initialize the volumeMigrationService
+		volumeMigrationService, err = initVolumeMigrationService(ctx, metadataSyncer)
```

Comment:
Is this needed? This function is only called in a full sync cycle, which initializes VMS at the start of the cycle..

Reply:
Makes sense. It's unnecessary